.github/actions/sign_and_upload_package/action.yml

```diff
@@ -0,0 +1,71 @@
+name: Sign and Upload Package
+description: 'Signs native modules with garasign'
+
+inputs: 
+    aws_role_arn:
+      description: 'AWS role input for drivers-github-tools/gpg-sign@v2'
+      required: true
+    aws_region_name:
+      description: 'AWS region name input for drivers-github-tools/gpg-sign@v2'
+      required: true
+    aws_secret_id:
+      description: 'AWS secret id input for drivers-github-tools/gpg-sign@v2'
+      required: true
+    npm_package_name:
+      description: 'The name for the npm package this repository represents'
+      required: true
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/download-artifact@v4
+
+    - name: Make signatures directory
+      shell: bash
+      run: mkdir artifacts
+
+    - name: Set up drivers-github-tools
+      uses: mongodb-labs/drivers-github-tools/setup@v2
+      with: 
+        aws_region_name: ${{ inputs.aws_region_name }}
+        aws_role_arn: ${{ inputs.aws_role_arn }}
+        aws_secret_id: ${{ inputs.aws_secret_id }}
+
+    - name: Create detached signature
+      uses: mongodb-labs/drivers-github-tools/gpg-sign@v2
+      with: 
+        filenames: 'build-*/*.tar.gz'
+      env: 
+        RELEASE_ASSETS: artifacts/
+
+    - name: Copy the tarballs to the artifacts directory
+      shell: bash
+      run: for filename in build-*/*.tar.gz; do cp ${filename} artifacts/; done
+
+    - run: npm pack
+      shell: bash
+
+    - name: Get release version and release package file name
+      id: get_vars
+      shell: bash
+      run: |
+        package_version=$(jq --raw-output '.version' package.json)
```

Comment:
code LGTM, is is possible to point me to an example test release, as an extra check?

Reply:
What I'm going off is seeing release-please creating the PR now: https://github.com/mongodb-js/mongodb-client-encryption/pull/15 and was verifying the signing actions working but not being able to upload before release-please was put in, example run here: https://github.com/mongodb-js/mongodb-client-encryption/actions/runs/9403489046/job/25900735594

If that's not sufficient let me know and I can create a new fake repo and project and copy everything over.

Reply:
No need for a fork: Can we have a run where we run `build.yml` to see the structure of the downloaded files, but comment out `gh release upload ..` and `npm publish`? To confirm the signatures are there.